src/x11/event.rs

```diff
@@ -47,6 +47,7 @@ derive_xrb! {
 	/// [`KEY_PRESS`]: crate::mask::EventMask::KEY_PRESS
 	pub struct KeyPress: Event(2) {
 		#[sequence]
+		#[hide(PartialEq, Hash)]
```

Comment:
I don't know, i found it a bit weird to have attributes before the documentation, since it separates the actual code, and you have to look at 2 different locations to have all informations on the field you looking at.
I would rather put the attributes and the field close to each other, like this:
```rust
struct Thing {
    /// Doc
    ///
    ///  # Things
    ///
    /// Lot of details...
    #[important_attribute]
    #[other_less_fancy_attribute]
    field: Type,
}
````

Reply:
Hmm.  For me that's more difficult to read because it separates the documentation from the code a bit more and makes the important attributes blend in, or take attention away from the docs.

However, the fact that you find that easier means there's probably a better way where it's easier for everyone to read.  Which is obviously the best way.  Not sure what that way is yet.

Reply:
> I don't know, i found it a bit weird to have attributes before the documentation, since it separates the actual code, and you have to look at 2 different locations to have all informations on the field you looking at. I would rather put the attributes and the field close to each other, like this:
> 
> ```rust
> struct Thing {
>     /// Doc
>     ///
>     ///  # Things
>     ///
>     /// Lot of details...
>     #[important_attribute]
>     #[other_less_fancy_attribute]
>     field: Type,
> }
> ```

We will stick to docs before attributes from now on.